unique.php

```diff
@@ -2931,6 +3091,107 @@ function ewww_image_optimizer_install_pngout() {
 	return $sendback;
 }
 
+/**
+ * Redirects back to previous page after SVGCLEANER installation.
+ */
+function ewww_image_optimizer_install_svgcleaner_wrapper() {
+	ewwwio_debug_message( '<b>' . __FUNCTION__ . '()</b>' );
+	if ( ! current_user_can( apply_filters( 'ewww_image_optimizer_admin_permissions', '' ) ) ) {
+		wp_die( esc_html__( 'You do not have permission to install image optimizer utilities.', 'ewww-image-optimizer' ) );
+	}
+	$sendback = ewww_image_optimizer_install_svgcleaner();
+	wp_safe_redirect( $sendback );
+	ewwwio_memory( __FUNCTION__ );
+	exit( 0 );
+}
+
+/**
+ * Installs svgcleaner from the official site.
+ *
+ * @return string The url from whence we came (settings page), with success or error parameters added.
+ */
+function ewww_image_optimizer_install_svgcleaner() {
+	if ( ! extension_loaded( 'zlib' ) || ! class_exists( 'PharData' ) ) {
+		$download_error = __( 'zlib or phar extension missing from PHP', 'ewww-image-optimizer' );
+	}
+	$os_chmod  = true;
+	$os_binary = 'svgcleaner';
+	if ( PHP_OS === 'Linux' ) {
+		$arch_type = 'x86_64';
+		if ( ewww_image_optimizer_function_exists( 'php_uname' ) ) {
+			$arch_type = php_uname( 'm' );
+		}
+		$os_string = 'linux_' . $arch_type;
+	} elseif ( PHP_OS === 'Darwin' ) {
+		$os_string = 'macos';
+	} elseif ( PHP_OS === 'WINNT' ) {
+		$os_chmod  = false;
+		$os_string = 'win32';
+		$os_binary = 'svgcleaner.exe';
+	}
+	$latest    = '0.9.5';
+	$tool_path = trailingslashit( EWWW_IMAGE_OPTIMIZER_TOOL_PATH );
+	if ( empty( $download_error ) ) {
+		$download_result = download_url( 'https://github.com/RazrFalcon/svgcleaner/releases/download/v' . $latest . '/svgcleaner_' . $os_string . '_' . $latest . '.tar.gz' );
```

Comment:
The Mac OS and Windows versions are in zip files, I don't see that we have a way to download those. The extension ought to be dynamic then.

Reply:
fixed, added zip extension for macos/win